src/rebuild/srv.c

```diff
@@ -1069,6 +1069,53 @@ rebuild_task_destroy(struct rebuild_task *task)
 	D_FREE(task);
 }
 
+/* Try merge the tasks to the current task.
+ *
+ * return 1 means merging the rebuild target to other task.
+ * return 0 means these targets can not merge.
+ *
+ **/
+static int
+rebuild_try_merge_tgts(const uuid_t pool_uuid, uint32_t map_ver,
```

Comment:
Instead of passing and merging failed target, and play with pool map version at the same time, I think it'll be simpler to identify each rebuild by 'pool map version' only, and each round rebuild will always rebuild the DOWN targets with fseq <= dst_map_ver.

Reply:
I do not see much difference here. In current implementation, it also tries to add all targets to 1 rebuild task, then do rebuild.